src/util/commandUtils.ts

```diff
@@ -0,0 +1,251 @@
+            }
+            return "\"" + gradleCmdStart + "\" " + `${command}` + ` -b="${buildGradlePath}"`; //wsl
+        default:
+            gradleCmdStart = Path.join(gradleCmdStart, "gradlew.bat");
+            if (customCommand) {
+                "\"" + gradleCmdStart + "\" " + `${command}` + ` ${customCommand}` + ` -b="${buildGradlePath}"`;
+            }
+            return "\"" + gradleCmdStart + "\" " + `${command}` + ` -b="${buildGradlePath}"`;
+    }
+}
+/**
+ * Reused from vscode-maven
+ * https://github.com/microsoft/vscode-maven/blob/2ab8f392f418c8e0fe2903387f2b0013a1c50e78/src/utils/mavenUtils.ts
+ */
+export function isWin(): boolean {
+    return process.platform.startsWith("win");
+}
+
+
+// return Maven executable path, Maven wrapper, or mvn
+async function mvnCmd(pomPath: string): Promise<string> {
+    const preferMavenWrapper: boolean | undefined = vscode.workspace.getConfiguration("maven").get<boolean>("executable.preferMavenWrapper");
+    if (preferMavenWrapper) {
+        const localMvnwPath: string | undefined = await getLocalMavenWrapper(Path.dirname(pomPath));
+        if (localMvnwPath) {
+            return `${localMvnwPath}`;
+        }
+    }
+    return "mvn";
+}
+
+async function gradleCmd(buildGradle: string): Promise<string> {
+    const preferGradleWrapper: boolean | undefined = vscode.workspace.getConfiguration("java").get<boolean>("import.gradle.wrapper.enabled");
+    if (preferGradleWrapper) {
+        const localGradlewPath: string | undefined = await getLocalGradleWrapper(Path.dirname(buildGradle));
+        if (localGradlewPath) {
+            return `${localGradlewPath}`;
+        }
+    }
+    return "gradle";
+}
+/**
+ * Search for potential Maven wrapper, return undefined if does not exist
+ * Reused from vscode-maven
+ * https://github.com/microsoft/vscode-maven/blob/2ab8f392f418c8e0fe2903387f2b0013a1c50e78/src/utils/mavenUtils.ts
+ * @param projectFolder
+ */
+async function getLocalMavenWrapper(projectFolder: string): Promise<string | undefined> {
+
+    let current: string = projectFolder;
+    while (Path.basename(current)) {
+        const potentialMvnwPath: string = Path.join(current);
+
+        if (await pathExists(potentialMvnwPath)) {
+            return potentialMvnwPath;
+        }
+        current = Path.dirname(current);
+    }
+    return undefined;
+}
+/**
+ * Search for potential Gradle wrapper, return undefined if it does not exist
+ * Modified from vscode-maven, see getLocalMavenWrapper method above
+ * @param projectFolder
+ */
+async function getLocalGradleWrapper(projectFolder: string): Promise<string | undefined> {
+
+    let current: string = projectFolder;
+    while (Path.basename(current)) {
+        const potentialGradlewPath: string = Path.join(current);
+        if (await pathExists(potentialGradlewPath)) {
+            return potentialGradlewPath;
+        }
+        current = Path.dirname(current);
+    }
+    return undefined;
+}
```

Comment:
Have you tested various scenarios with Maven and Gradle wrappers? I do not think these methods will work as intended now that you have removed these lines:
`const mvnw: string = isWin() ? "mvnw.cmd" : "mvnw";`
and
`const gradlew: string = isWin() ? "gradlew.bat" : "gradlew";`

The whole point behind these methods was to test that the wrapper actually exists. Without the `mvnw` or `gradlew` appended to the folder paths passed to these functions, all the method is doing is verifying that the passed in path exists, not verifying that there is a Maven or Gradle wrapper at that path.

Reply:
Thanks for the clarification. Made the changes as you suggested , the scenarios with Maven and Gradle wrappers were working fine before and after the changes.

Reply:
> the scenarios with Maven and Gradle wrappers were working fine before and after the changes.
That would only be the case if you tested with Maven/Gradle wrappers located in the project folder path that was initially passed in. If the wrapper is located in a parent folder, then the changes you made would not have worked.